virtcontainers/network.go

```diff
@@ -1110,11 +1118,14 @@ func createEndpoint(netInfo NetworkInfo, idx int, model NetInterworkingModel) (E
 		if socketPath != "" {
 			networkLogger().WithField("interface", netInfo.Iface.Name).Info("VhostUser network interface found")
 			endpoint, err = createVhostUserEndpoint(netInfo, socketPath)
+		} else if model == NetXConnectNoneModel {
```

Comment:
That's not right. We cannot rely on the internetworking model to decide we want to create a tap network endpoint.
Instead we should rely on:
```go
} else if netInfo.Iface.Type == TapEndpointType {
```
The internetworking model should only be used to understand how the created endpoint (`tap` in this case) is supposed to be connected to the host network.

Reply:
This reminds me one thing. In hotplug network https://github.com/kata-containers/runtime/blob/d00742f43f3fb1177e74d864e47a1bae697c2d73/virtcontainers/sandbox.go#L1075-L1099, `netInfo.Iface.Type` is always "". So actually hotplug doesn't work now. You mean we set type there?